core/unit_test/TestNestedReducerCTAD.cpp

```diff
@@ -0,0 +1,248 @@
+          Kokkos::MinFirstLoc<ScalarType, IndexType, MemorySpace>;
+      using ValueType = ReducerTypeExpected::value_type;
+      Kokkos::View<ValueType, MemorySpace> view;
+      Kokkos::MinFirstLoc reducer(view);
+      check_types<ReducerTypeExpected>(team_handle, reducer);
+    }
+
+    {
+      using ReducerTypeExpected =
+          Kokkos::MinFirstLocCustomComparator<ScalarType, IndexType,
+                                              FakeComparator, MemorySpace>;
+      using ValueType = ReducerTypeExpected::value_type;
+      Kokkos::View<ValueType, MemorySpace> view;
+      FakeComparator comparator;
+      Kokkos::MinFirstLocCustomComparator reducer(view, comparator);
+      check_types<ReducerTypeExpected>(team_handle, reducer);
+    }
+
+    {
+      using ReducerTypeExpected =
+          Kokkos::MinMaxFirstLastLoc<ScalarType, IndexType, MemorySpace>;
+      using ValueType = ReducerTypeExpected::value_type;
+      Kokkos::View<ValueType, MemorySpace> view;
+      Kokkos::MinMaxFirstLastLoc reducer(view);
+      check_types<ReducerTypeExpected>(team_handle, reducer);
+    }
+
+    {
+      using ReducerTypeExpected = Kokkos::MinMaxFirstLastLocCustomComparator<
+          ScalarType, IndexType, FakeComparator, MemorySpace>;
+      using ValueType = ReducerTypeExpected::value_type;
+      Kokkos::View<ValueType, MemorySpace> view;
+      FakeComparator comparator;
+      Kokkos::MinMaxFirstLastLocCustomComparator reducer(view, comparator);
+      check_types<ReducerTypeExpected>(team_handle, reducer);
+    }
+
+    {
+      using ReducerTypeExpected = Kokkos::FirstLoc<IndexType, MemorySpace>;
+      using ValueType           = ReducerTypeExpected::value_type;
+      Kokkos::View<ValueType, MemorySpace> view;
+      Kokkos::FirstLoc reducer(view);
+      check_types<ReducerTypeExpected>(team_handle, reducer);
+    }
+
+    {
+      using ReducerTypeExpected = Kokkos::LastLoc<IndexType, MemorySpace>;
+      using ValueType           = ReducerTypeExpected::value_type;
+      Kokkos::View<ValueType, MemorySpace> view;
+      Kokkos::LastLoc reducer(view);
+      check_types<ReducerTypeExpected>(team_handle, reducer);
+    }
+
+    {
+      using ReducerTypeExpected =
+          Kokkos::StdIsPartitioned<IndexType, MemorySpace>;
+      using ValueType = ReducerTypeExpected::value_type;
+      Kokkos::View<ValueType, MemorySpace> view;
+      Kokkos::StdIsPartitioned reducer(view);
+      check_types<ReducerTypeExpected>(team_handle, reducer);
+    }
+
+    {
+      using ReducerTypeExpected =
+          Kokkos::StdPartitionPoint<IndexType, MemorySpace>;
+      using ValueType = ReducerTypeExpected::value_type;
+      Kokkos::View<ValueType, MemorySpace> view;
+      Kokkos::StdPartitionPoint reducer(view);
+      check_types<ReducerTypeExpected>(team_handle, reducer);
+    }
+  }
+
+  static void test_ctad_inside_parallel_for() {
+    Kokkos::parallel_for(TeamPolicy(0, Kokkos::AUTO), TestNestedReducerCTAD());
```

Comment:
Do we need the parallel_for?  Are we not able reproduce the issue by simply using CTAD within a KOKKOS_FUNCTION?

Reply:
If you don't compile with RDC I think it would not necessarily compile the function depending on the optimization level

Reply:
Didn't you manage to reproduce on the compiler explorer?  Did you actually write a kernel launch?

Reply:
If I put the same code in an anonymous namespace, the compiler does not generate any assembly, https://godbolt.org/z/8fxEnM6P1.

Do you prefer that I remove the anonymous namespace and make a dedicated executable for this test ?

Reply:
We don't care about assembly being generated.  Don't.

Reply:
I meant the compiler does not generate any error if I remove the call to `parallel_for`.

Reply:
OK.  Then withdrawn.